contracts/RewardController.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+// Disclaimer https://github.com/hats-finance/hats-contracts/blob/main/DISCLAIMER.md
+
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "./HATVaults.sol";
+
+contract RewardController is OwnableUpgradeable {
+
+    struct PoolUpdate {
+        uint256 blockNumber;// update blocknumber
+        uint256 totalAllocPoint; //totalAllocPoint
+    }
+
+    uint256 public constant MULTIPLIERS_LENGTH = 24;
+
+    uint256 public rewardPerBlock;
+    // Block from which the vaults contract will start rewarding.
+    uint256 public startBlock;
+    uint256 public epochLength;
+    // Reward Multipliers
+    uint256[24] public rewardPerEpoch;
+    PoolUpdate[] public globalPoolUpdates;
+    mapping(uint256 => uint256) public poolsAllocPoint;
+    mapping(uint256 => uint256) public poolsLastProcessedTotalAllocPoint;
+    HATVaults public hatVaults;
+
+    event SetRewardPerEpoch(uint256[24] _rewardPerEpoch);
+
+    modifier onlyVaults() {
+        require(msg.sender == address(hatVaults), "Only vaults");
+        _;
+    }
+
+    function initialize(
+        address _hatGovernance,
+        HATVaults _hatVaults,
+        uint256 _rewardPerBlock,
+        uint256 _startRewardingBlock,
+        uint256 _epochLength
+    ) external initializer {
+        rewardPerBlock = _rewardPerBlock;
+        startBlock = _startRewardingBlock;
+        epochLength = _epochLength;
+        hatVaults = _hatVaults;
+        _transferOwnership(_hatGovernance);
+
+        rewardPerEpoch = [
+            4413, 4413, 8825, 7788, 6873, 6065,
+            5353, 4724, 4169, 3679, 3247, 2865,
+            2528, 2231, 1969, 1738, 1534, 1353,
+            1194, 1054, 930, 821, 724, 639
+        ];
+    }
+
+    /**
+     * @dev setRewardPerEpoch- called by hats governance to set reward multipliers
+     * @param _rewardPerEpoch reward multipliers
+    */
+    function setRewardPerEpoch(uint256[24] memory _rewardPerEpoch) external onlyOwner {
+        rewardPerEpoch = _rewardPerEpoch;
+        emit SetRewardPerEpoch(_rewardPerEpoch);
+    }
+
+    function setAllocPoints(uint256 _pid, uint256 _allocPoint) external onlyOwner {
```

Comment:
> or mb we need a setter to set a batch?

when it's done by the voters it won't be necessary, but maybe for now could be useful

Reply:
on the contrary, the way it will work "when it is done by the voters" is  (probably) that all alloc points will all be update together at the same time :)

anyway, just rename to setAllocPoint, we can think of that later